Sources/PenguinTesting/StdlibProtocolTests.swift

```diff
@@ -110,24 +110,37 @@ extension Comparable {
   /// different as possible internally, while still being equal.  Otherwise, it's fine to pass `nil`
   /// (the default) for `self1` and `self2`.
   ///
-  /// - Precondition: `self == (self1 ?? self) && self1 == (self2 ?? self)`
-  /// - Precondition: `self < greater && greater < greaterStill`.
+  /// If the caller does not have enough example values for `greater` or `greaterStill`, then
+  /// the caller may pass `nil` values. Callers are encouraged to pass non-`nil` values whenever
+  /// they are available, because they enable more checks.
+  ///
+  /// - Precondition: `self == (self1 ?? self) && self1 == (self2 ?? self)`.
```

Comment:
Calling these preconditions is questionable at best, IMO.  Should code that fails a test be considered illegal?  I don't think so. At some point we may want to be able to even test that a given test fails.

Reply:
Most of these preconditions were already here. I just updated them to take into account the new optionality. And the tests themselves do have `precondition` statements for all of them.

Maybe this is something we should revisit later?